devel/onetbb/files/patch-onetbb-dispatch-fallback.diff

```diff
@@ -0,0 +1,100 @@
+diff --git src/tbb/semaphore.h src/tbb/semaphore.h
+index 9d27f3ac..b8942329 100644
+--- src/tbb/semaphore.h
++++ src/tbb/semaphore.h
+@@ -22,8 +22,16 @@
+ #if _WIN32||_WIN64
+ #include <windows.h>
+ #elif __APPLE__
++#include <AvailabilityMacros.h>
++#if MAC_OS_X_VERSION_MIN_REQUIRED > 1060
+ #include <dispatch/dispatch.h>
```

Comment:
Right, why is it being bypassed?

Reply:
Lumme amend the check to may it correct.

Reply:
but the 1050 sdk does not have libdispatch, so it should not be used there.

https://github.com/alexey-lysiuk/macos-sdk/tree/main/MacOSX10.5.sdk/usr/include

I understand what you’re trying to do, but all this monkeying around trying to make sure 10.6-PPC support is slipped in is breaking things. It’s just too hard to always get it right.

Reply:
the version numbers can be (and are sometimes) set to patch levels, like 1058 for 10.5

https://trac.macports.org/wiki/LeopardSDKFixes

so it’s usually done as

>= 1060

if you mean 10.6 and above.

It would be great if libdispatch could be made to work on older systems. However, I thought it needed kernel support to function. I can take a look at the port you set up and see if it functions.

Reply:
@kencu Thank you, I keep forgetting about patch levels. You are right, it should be fixed.

My port at the moment will not build on Leopard. I think kernel is not involved though, since FreeBSD ported GCD. (This is no more than a guess.)
There are two immediate issues:
1. Something fails with ObjC code, so either 10.5 gcc is at fault or some headers needed.
2. There are a few functions from Libc (and libpthreads maybe) which are missing.
I used Xcode to build, however it also supports makefiles. Maybe worth trying.

If you could take a look at that, would be awesome. The issue is nothing ppc-specific and can be addressed on x86 just as well.

Reply:
I will also look into that again. What could be worth trying is disabling linking against system libs, maybe in that case the build system pulls in what is needed (and missing on 10.5) from supplied sources.
Not sure about gcc, but we could try Plausible blocks compiler. Installing it is trivial, and I think it can be enabled without tweaking the base.

Reply:
> 2. There are a few functions from Libc (and libpthreads maybe) which are missing.

this likely relates to the kernel support I read about way back when, eg

https://stackoverflow.com/questions/1581776/how-does-grand-central-dispatch-really-use-the-operating-system

https://newosxbook.com/articles/GCD.html